### DISCUSSION TITLE
Data analysis strategy

### DISCUSSION THREAD

**Discussion by Simon-Bertrand** [Simon-Bertrand/challengedata.ens.fr-RealEstatePricePrediction#1] [Ideas]
-Description of features - Simon
1. Categorical  : total number of category, proportion of missing values, proportion of each category
2. Continuous :  min, max, mean, median, quartiles, std, proportion of missing values
- [X] Prototype : getDescriptor(pd.data) -> dict : 1 key / feature : categorical/continuous object descriptor

-1-D distribution plotter (continuous data) - Léo
- [x] Prototype : showDistribPlot(pd.data, column : str) -> matplotlib / plotly /seaborn object

-1-D categorical proportion plotter (categorical data) - Léo
- [x] Prototype : showDistribProp(pd.data, column : str) -> matplotlib / plotly /seaborn object

-Correlation matrix plot (continuous values) - Léo
- [x] Prototype : showCorrelationMatrix(pd.data) -> matplotlib / plotly /seaborn object

-Pair plot (continuous values) - Léo
- [ ] Prototype : showPairPlot(pd.data) -> matplotlib / plotly /seaborn object

(Ref :
 https://seaborn.pydata.org/tutorial/categorical.html
https://seaborn.pydata.org/tutorial/distributions.html
)

**Comment by LeoLoehrer** [comment]
- Missing values properties


- if not property_type in ["appartement", "chambre", "duplex"] => floor == NaN 

- if property_type in ["terrain", "terrain à bâtir"] :
        - if Size != NaN and land_size !=NaN and size!=land_size => property_type misclassified (habitation) (14)
        - elif Size != NaN or land_size != NaN => size == land_size (1188)
        - elif size == NaN and land_size == NaN => cannot process a regression ? (51)
   
- if property_type == 'appartement' => land_size = NaN 95,8% du temps (15 087 /15 733)


Source:
![image](https://user-images.githubusercontent.com/84455908/215775455-22105010-66c0-4936-bca4-10b0c020945c.png)

![image](https://user-images.githubusercontent.com/84455908/215812230-82535842-3004-4eed-a98f-430aaea3058b.png)

**Comment by Simon-Bertrand** [comment]
Data report :

_________________________
property_type:
[numberOfCategories] = 22 ; 
[proportionOfEachCategory] = {'appartement': 0.42102868764718476, 'atelier': 5.352172982230786e-05, 'chalet': 0.0012845215157353885, 'chambre': 0.00010704345964461572, 'château': 0.0005619781631342324, 'divers': 0.057187968315135944, 'duplex': 0.005887390280453864, 'ferme': 0.003478912438450011, 'gîte': 8.028259473346178e-05, 'hôtel': 2.676086491115393e-05, 'hôtel particulier': 0.00016056518946692356, 'loft': 0.001445086705202312, 'maison': 0.4194765574823378, 'manoir': 0.0004014129736673089, 'moulin': 0.00045493470348961676, 'parking': 0.002569043031470777, 'propriété': 0.010918432883750802, 'péniche': 0.00021408691928923143, 'terrain': 0.041051166773710124, 'terrain à bâtir': 0.00797473774352387, 'viager': 0.004736673089274245, 'villa': 0.02090023549561122} ; 
[missingValues] = 0 ; 
[col_type] = object ; 

_________________________
approximate_latitude:
[min] = 41.3744 ; 
[max] = 51.0759 ; 
[mean] = 46.5462 ; 
[median] = 46.9394 ; 
[quartile_1] = 43.9215 ; 
[quartile_3] = 48.8415 ; 
[std] = 2.3566 ; 
[missingValues] = 0 ; 
[col_type] = float64 ; 

_________________________
approximate_longitude:
[min] = -4.7335 ; 
[max] = 9.5032 ; 
[mean] = 2.6191 ; 
[median] = 2.3808 ; 
[quartile_1] = 1.096 ; 
[quartile_3] = 4.6242 ; 
[std] = 2.5971 ; 
[missingValues] = 0 ; 
[col_type] = float64 ; 

_________________________
city:
[numberOfCategories] = 8643 ; 
[missingValues] = 0 ; 
[col_type] = object ; 

_________________________
size:
[min] = 1.0 ; 
[max] = 411311.0 ; 
[mean] = 1088.8316 ; 
[median] = 115.0 ; 
[quartile_1] = 74.0 ; 
[quartile_3] = 235.25 ; 
[std] = 5555.3209 ; 
[missingValues] = 512 ; 
[col_type] = float64 ; 

_________________________
floor:
[min] = 1.0 ; 
[max] = 55.0 ; 
[mean] = 3.4795 ; 
[median] = 2.0 ; 
[quartile_1] = 1.0 ; 
[quartile_3] = 4.0 ; 
[std] = 6.7256 ; 
[missingValues] = 27625 ; 
[col_type] = float64 ; 

_________________________
land_size:
[min] = 1.0 ; 
[max] = 6203700.0 ; 
[mean] = 3995.6654 ; 
[median] = 795.0 ; 
[quartile_1] = 362.0 ; 
[quartile_3] = 1841.0 ; 
[std] = 54415.9541 ; 
[missingValues] = 21787 ; 
[col_type] = float64 ; 

_________________________
energy_performance_value:
[min] = 0.0 ; 
[max] = 100000.0 ; 
[mean] = 205.3851 ; 
[median] = 180.0 ; 
[quartile_1] = 125.0 ; 
[quartile_3] = 239.0 ; 
[std] = 790.5018 ; 
[missingValues] = 18300 ; 
[col_type] = float64 ; 

_________________________
energy_performance_category:
[numberOfCategories] = 7 ; 
[proportionOfEachCategory] = {'A': 0.01648469278527082, 'B': 0.0572147291800471, 'C': 0.11006743737957611, 'D': 0.18794155427103404, 'E': 0.10203917790622993, 'F': 0.02807214729180047, 'G': 0.008456433311924642} ; 
[missingValues] = 18300 ; 
[col_type] = object ; 

_________________________
ghg_value:
[min] = 0.0 ; 
[max] = 17024.0 ; 
[mean] = 31.8455 ; 
[median] = 16.0 ; 
[quartile_1] = 8.0 ; 
[quartile_3] = 36.0 ; 
[std] = 310.5762 ; 
[missingValues] = 18838 ; 
[col_type] = float64 ; 

_________________________
ghg_category:
[numberOfCategories] = 7 ; 
[proportionOfEachCategory] = {'A': 0.06815992292870905, 'B': 0.12125347891243846, 'C': 0.0915489188610576, 'D': 0.0868925283665168, 'E': 0.08049668165275102, 'F': 0.03401305930207664, 'G': 0.013514236780132733} ; 
[missingValues] = 18838 ; 
[col_type] = object ; 

_________________________
exposition:
[numberOfCategories] = 12 ; 
[proportionOfEachCategory] = {'Est': 0.018518518518518517, 'Est-Ouest': 0.021943909227146223, 'Nord': 0.006181759794476558, 'Nord-Est': 0.005914151145365018, 'Nord-Ouest': 0.008536715906658103, 'Nord-Sud': 0.0006690216227788482, 'Ouest': 0.02159601798330122, 'Ouest-Est': 0.001070434596446157, 'Sud': 0.08667844144722757, 'Sud-Est': 0.025583386855063155, 'Sud-Nord': 0.009339541853992722, 'Sud-Ouest': 0.03733140655105973} ; 
[missingValues] = 28274 ; 
[col_type] = object ; 

_________________________
nb_rooms:
[min] = 0.0 ; 
[max] = 130.0 ; 
[mean] = 4.2454 ; 
[median] = 4.0 ; 
[quartile_1] = 3.0 ; 
[quartile_3] = 5.0 ; 
[std] = 2.9878 ; 
[missingValues] = 1566 ; 
[col_type] = float64 ; 

_________________________
nb_bedrooms:
[min] = 0.0 ; 
[max] = 118.0 ; 
[mean] = 2.8649 ; 
[median] = 3.0 ; 
[quartile_1] = 2.0 ; 
[quartile_3] = 4.0 ; 
[std] = 2.1565 ; 
[missingValues] = 2733 ; 
[col_type] = float64 ; 

_________________________
nb_bathrooms:
[min] = 0.0 ; 
[max] = 3.0 ; 
[mean] = 0.9207 ; 
[median] = 1.0 ; 
[quartile_1] = 1.0 ; 
[quartile_3] = 1.0 ; 
[std] = 0.2708 ; 
[missingValues] = 13273 ; 
[col_type] = float64 ; 

_________________________
nb_parking_places:
[min] = 0.0 ; 
[max] = 1.0 ; 
[mean] = 0.2935 ; 
[median] = 0.0 ; 
[quartile_1] = 0.0 ; 
[quartile_3] = 1.0 ; 
[std] = 0.4554 ; 
[missingValues] = 0 ; 
[col_type] = float64 ; 

_________________________
nb_boxes:
[min] = 0.0 ; 
[max] = 1.0 ; 
[mean] = 0.1797 ; 
[median] = 0.0 ; 
[quartile_1] = 0.0 ; 
[quartile_3] = 0.0 ; 
[std] = 0.3839 ; 
[missingValues] = 0 ; 
[col_type] = float64 ; 

_________________________
nb_photos:
[min] = 1.0 ; 
[max] = 50.0 ; 
[mean] = 7.9792 ; 
[median] = 8.0 ; 
[quartile_1] = 5.0 ; 
[quartile_3] = 10.0 ; 
[std] = 4.6461 ; 
[missingValues] = 0 ; 
[col_type] = float64 ; 

_________________________
has_a_balcony:
[min] = 0.0 ; 
[max] = 1.0 ; 
[mean] = 0.1483 ; 
[median] = 0.0 ; 
[quartile_1] = 0.0 ; 
[quartile_3] = 0.0 ; 
[std] = 0.3554 ; 
[missingValues] = 0 ; 
[col_type] = float64 ; 

_________________________
nb_terraces:
[min] = 0.0 ; 
[max] = 1.0 ; 
[mean] = 0.3096 ; 
[median] = 0.0 ; 
[quartile_1] = 0.0 ; 
[quartile_3] = 1.0 ; 
[std] = 0.4623 ; 
[missingValues] = 0 ; 
[col_type] = float64 ; 

_________________________
has_a_cellar:
[min] = 0.0 ; 
[max] = 1.0 ; 
[mean] = 0.2001 ; 
[median] = 0.0 ; 
[quartile_1] = 0.0 ; 
[quartile_3] = 0.0 ; 
[std] = 0.4001 ; 
[missingValues] = 0 ; 
[col_type] = float64 ; 

_________________________
has_a_garage:
[min] = 0.0 ; 
[max] = 1.0 ; 
[mean] = 0.0539 ; 
[median] = 0.0 ; 
[quartile_1] = 0.0 ; 
[quartile_3] = 0.0 ; 
[std] = 0.2258 ; 
[missingValues] = 0 ; 
[col_type] = float64 ; 

_________________________
has_air_conditioning:
[min] = 0.0 ; 
[max] = 1.0 ; 
[mean] = 0.0398 ; 
[median] = 0.0 ; 
[quartile_1] = 0.0 ; 
[quartile_3] = 0.0 ; 
[std] = 0.1956 ; 
[missingValues] = 0 ; 
[col_type] = float64 ; 

_________________________
last_floor:
[min] = 0.0 ; 
[max] = 1.0 ; 
[mean] = 0.0038 ; 
[median] = 0.0 ; 
[quartile_1] = 0.0 ; 
[quartile_3] = 0.0 ; 
[std] = 0.0617 ; 
[missingValues] = 0 ; 
[col_type] = float64 ; 

_________________________
upper_floors:
[min] = 0.0 ; 
[max] = 1.0 ; 
[mean] = 0.0002 ; 
[median] = 0.0 ; 
[quartile_1] = 0.0 ; 
[quartile_3] = 0.0 ; 
[std] = 0.0155 ; 
[missingValues] = 0 ; 
[col_type] = float64 ; 

_________________________
price:
[min] = 24465.0 ; 
[max] = 2299000.0 ; 
[mean] = 343221.2721 ; 
[median] = 255250.0 ; 
[quartile_1] = 155462.5 ; 
[quartile_3] = 415000.0 ; 
[std] = 308912.9143 ; 
[missingValues] = 0 ; 
[col_type] = float64 ;

**Comment by Simon-Bertrand** [comment]
Selected columns for preprocessing : 
['property_type', 'city', 'size', 'floor', 'land_size', 'nb_rooms',
       'nb_bedrooms', 'nb_bathrooms', 'nb_parking_places', 'nb_boxes',
       'has_a_balcony', 'nb_terraces', 'has_a_cellar', 'has_a_garage',
       'has_air_conditioning', 'last_floor', 'upper_floors', 'price',
       'images']
